public/js/subcircuit.js

```diff
@@ -190,6 +197,11 @@ SubCircuit.prototype.reset = function() {
         return;
     }
 
+    if(allLayouts[subcircuitScope.id]){
+        // Set the width and height
+        subcircuitScope.layout = Object.assign(allLayouts[subcircuitScope.id]);
+    }
+
     subcircuitScope.layout.height = subcircuitScope.layout.height;
```

Comment:
These two lines seem like a no-op. I wonder if there was a typo in the original code.
Maybe we don't need the allLayouts copy if from here we could get a hold to the updated subcircuit template.
I was looking for a pointer to it but I'm not familiar with this part yet, @satu0king , do you know which variable would be holding all the loaded circuits here?

Reply:
Ah, I did some investigation:

By the end of saveLayout() the globalScope.layout is correct (and it is the same as scopeList[<id>].layout). As we switch circuits, by the time SubCircuit.reset is called the scopeList[<id>].layout.height was already reset.

Looking for where it happens I found switchCircuit() calling Scope.prototype.fixLayout(); this is the one resetting the height.

I think we can fix the issue by just removing Scope.prototype.fixLayout (in logix.js) altogether and its call from switchCircuit() (in data.js).